timesketch/api/v1/resources.py

```diff
@@ -1603,24 +1582,188 @@ def post(self):
                 db_session.add(timeline)
                 db_session.commit()
 
-        enable_stream = form.enable_stream.data
         # Start Celery pipeline for indexing and analysis.
         # Import here to avoid circular imports.
         # pylint: disable=import-outside-toplevel
         from timesketch.lib import tasks
         pipeline = tasks.build_index_pipeline(
-            file_path, timeline_name, index_name, file_extension, sketch_id,
-            only_index=enable_stream)
+            file_path=file_path, events=events, timeline_name=timeline_name,
+            index_name=index_name, file_extension=file_extension,
+            sketch_id=sketch_id, only_index=enable_stream)
         pipeline.apply_async()
 
         # Return Timeline if it was created.
         # pylint: disable=no-else-return
         if timeline:
             return self.to_json(
-                timeline, status_code=HTTP_STATUS_CODE_CREATED)
+                timeline, status_code=HTTP_STATUS_CODE_CREATED, meta=meta)
 
         return self.to_json(
-            searchindex, status_code=HTTP_STATUS_CODE_CREATED)
+            searchindex, status_code=HTTP_STATUS_CODE_CREATED, meta=meta)
+
+    def _upload_events(self, events, form, sketch, index_name):
+        """Upload a file like object.
+
+        Args:
+            events: string with all the events.
+            form: a dict with the configuration for the upload.
+            sketch: Instance of timesketch.models.sketch.Sketch
+            index_name: the Elastic index name for the timeline.
+
+        Returns:
+            A timeline if created otherwise a search index in JSON (instance
+            of flask.wrappers.Response)
+        """
+        timeline_name = form.get('name', 'unknown_events')
+        file_extension = 'jsonl'
```

Comment:
this only applies to .jsonl files?

Reply:
no... you are adding events, and events need to be JSON. This is a way of telling the uploader that the events in question are json line formatted, not csv formatted. since that' show events are uploaded.